File: array_and_hashing/swift/lengthOfLastWord.swift
func lengthOfLastWord(_ s: String) -> Int {
    var s = s.split(separator: " ") // creates an array of substrings
    var lastWord = ""
    for i in stride(from: s.count - 1, through: 0, by: -1){
        if(s[i].count > 0){
            lastWord = String(s[i])
            break
        }
    }
    return lastWord.count
}

File: array_and_hashing/cpp/lengthOfLastWord.cpp
#include <string>

int lengthOfLastWord(std::string s) {
    int res = 0;
    int i = s.size() - 1;
    for(; i > -1; i--){
        if(s[i] != ' '){
            break;
        }
    }
    for(; i > -1; i--){
        if(s[i] == ' '){
            break;
        }
        res++;
    }
    return res;
}